pkg/controller/container-runtime-config/helpers.go

```diff
@@ -954,3 +1006,67 @@ func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature
 	}
 	return data, nil
 }
+
+func addScopeMirrorsSigstoreRegistriesdConfig(registriesDockerConfig map[string]dockerConfig, scope string, reg *sysregistriesv2.Registry, sigstoreAttachment dockerConfig) error {
+	if reg == nil {
+		return nil
+	}
+
+	dummyDigest := "@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+	dummyTag := ":aaa"
+	dummyPath := "/a"
+	dummyPrefix := "matched"
+
+	if strings.HasPrefix(scope, "*.") {
+		scope = strings.Replace(scope, "*", dummyPrefix, 1)
+	}
+
+	scopeRef, err := reference.Parse(scope)
+	if err != nil {
+		return fmt.Errorf("error parsing scope %s: %w", scope, err)
+	}
+
+	if digestRef, digested := scopeRef.(reference.Digested); digested {
+		namedRef, err := reference.ParseNamed(digestRef.String())
+		if err != nil {
+			return fmt.Errorf("error parsing scope digested reference %s: %w", digestRef.String(), err)
+		}
+		scope = reference.TrimNamed(namedRef).String()
+
+	}
+	if tagRef, tagged := scopeRef.(reference.Tagged); tagged {
+		namedRef, err := reference.ParseNamed(tagRef.String())
+		if err != nil {
+			return fmt.Errorf("error parsing scope tagged reference %s: %w", tagRef.String(), err)
+		}
+		scope = reference.TrimNamed(namedRef).String()
+	}
+	if !strings.Contains(scope, "/") {
+		scope += dummyPath
+	}
+
+	digestRef, err := reference.ParseNamed(scope + dummyDigest)
+	if err != nil {
+		return fmt.Errorf("error parsing digest name for scope %s: %w", scope, err)
+	}
+	tagRef, err := reference.ParseNamed(scope + dummyTag)
+	if err != nil {
+		return fmt.Errorf("error parsing tag name for scope %s: %w", scope, err)
+	}
+	digestSources, err := reg.PullSourcesFromReference(digestRef)
+	if err != nil {
+		return fmt.Errorf("error getting digest sources for scope %s: %w", scope, err)
+	}
+	tagSources, err := reg.PullSourcesFromReference(tagRef)
+	if err != nil {
+		return fmt.Errorf("error getting tag sources for scope %s: %w", scope, err)
+	}
+	for _, s := range append(digestSources, tagSources...) {
+		endpoint := s.Endpoint.Location
+		if endpoint == "" {
+			endpoint = reg.Prefix
+		}
```

Comment:
AFAICS this should be `strings.TrimSuffix(s.Reference.Name(), dummyPathIfUsed)`; `s.Endpoint` is the “root” of the mirror, not the final remapped repo (consider an original `scope` of `example.com/foo/bar`, and a mirror defined for `example.com/foo`)